### PR TITLE
wasmate: indent blocks and fix tests.

### PR DESCRIPTION
**Pull request by ncbray**
I need to add comments before committing, doing the request now so I don't forget this branch.

### CODE REVIEW

**Comment by ncbray** [comment]
Ready to review, now.

File: prototype-wasmate/wasmate.py [hunk 1]
```diff
@@ -283,13 +283,17 @@ def __init__(self):
...
+        self.ends_at_label = {}
 
-    def push_label(self, label):
-        if label in self.block_labels:
-            self.block_labels[label] += 1
-        else:
-            self.block_labels[label] = 1
+    def close_block(self, label):
+        out.dedent()
+        #out.write_line(') # ' + label)
```
Comment (binji) [inline prototype-wasmate/wasmate.py]: remove?

File: prototype-wasmate/wasmate.py [hunk 2]
```diff
@@ -283,13 +283,17 @@ def __init__(self):
...
+        out.write_line(')')
+
+    def block_ends_at(self, block_label, end_label):
+        if end_label not in self.ends_at_label:
```
Comment (binji) [inline prototype-wasmate/wasmate.py]: I prefer using collections.defaultdict for this, but OK

File: prototype-wasmate/wasmate.py [hunk 3]
```diff
@@ -324,13 +331,23 @@ def handle_mnemonic(self, command, args):
...
+            # at is after the loop.
+            # This creates an inconsistency between LLVM and sexprs where LLVM
+            # declares "begin: loop $end ... end: ..." but sexprs declare
+            # "(loop $begin ...)", the nature of the argument to the "loop"
+            # command differs.
             out.write_line('(loop $' + self.current_label)
+            out.indent()
             assert len(self.expr_stack) == 0
-            self.push_label(args[0][1:]) # strip leading $
+            self.block_ends_at(self.current_label, args[0][1:])
```
Comment (binji) [inline prototype-wasmate/wasmate.py]: still might be nice to mention that [1:] strips the leading $

**Comment by binji** [comment]
Much nicer to read, thanks. lgtm w/ comments

**Pull request merged**